environment/providers/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,106 @@
+/*
+ Kubernetes environment reads settings by accessing Kubernetes APIs.
+*/
+
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/nutanix-cloud-native/prism-go-client/environment/types"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+)
+
+type provider struct {
+	secretInformer coreinformers.SecretInformer
+	prismEndpoint  NutanixPrismEndpoint
+}
+
+func (prov *provider) getCredentials(_ types.Topology) (*types.ApiCredentials, error) {
+	ref := prov.prismEndpoint.CredentialRef
+	secret, err := prov.secretInformer.Lister().Secrets(ref.Namespace).Get(ref.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse credentials in secret
+	credsData, ok := secret.Data["credentials"]
```

Comment:
Does the secretInformer take care of base64 decoding?

Reply:
It does. base64 encoding is done by YAML encoder for data of type []byte. This code has been heavily influenced by CCM. See https://github.com/nutanix-cloud-native/cloud-provider-nutanix/blob/87510dd51501f288332d126a6bd9f535729edc18/pkg/provider/nutanix_client.go#L63
Idea is to consolidate common logic like parsing of Nutanix credentials across repos.